src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c

```diff
@@ -1300,6 +1302,95 @@ static int get_temp_by_m_tag(struct xocl_xmc *xmc, char *m_tag)
 }
 
 /* Runtime clock scaling sysfs node */
+static ssize_t scaling_threshold_power_override_en_show(struct device *dev,
+	struct device_attribute *da, char *buf)
+{
+	struct xocl_xmc *xmc = platform_get_drvdata(to_platform_device(dev));
+	u32 val = 0;
+
+	if (!xmc->runtime_cs_enabled) {
+		if (!XMC_PRIVILEGED(xmc))
+			xocl_dbg(dev, "runtime clock scaling is not supported in non privileged mode\n");
+		else if (xmc->cs_on_ptfm)
+			xocl_warn(dev, "runtime clock scaling is not enabled\n");
+		else
+			xocl_warn(dev, "runtime clock scaling is not supported\n");
+		return sprintf(buf, "%d\n", val);
```

Comment:
these code repeatedly show up in multiple places. can we make a function for them?